### PR TITLE
OSSM-2203: sets up jobs for xns-informer 2.4

### PR DESCRIPTION
**Pull request by bartoszmajsak**

### CODE REVIEW

**Comment by yxun** [comment]
" could not resolve base image from ci/maistra-builder:2.4"
I think we haven't mapped a 2.4 builder image. Do we have it in our quay.io repo ?

**Comment by yxun** [comment]
I created a new PR : https://github.com/openshift/release/pull/33500
When that one is merged, we can use the 2.4 builder image in ci and then this PR can find the builder image in job runs.

**Comment by bartoszmajsak** [comment]
> I created a new PR : https://github.com/openshift/release/pull/33500
> When that one is merged, we can use the 2.4 builder image in ci and then this PR can find the builder image in job runs. 

Thanks a lot. I failed to find this reference.

**Comment by yxun** [comment]
level=error msg="Failed to set up dependencies. This might cause subsequent failures." error="failed to ensure imagestreamtags in cluster build01: failed waiting for imagestreamtag build-cache/maistra-xns-informer:maistra-2.4 to appear: timed out waiting for the condition" org=openshift repo=release
 
 The two maistra jobs passed. The above error looks like an infrastructure cache error. If you have fetch the latest changes from openshift/release master branch, you can try to run a `make update` command in your local branch. 
 And then check if there is any additional change will be auto generated in the 
 ci-operator/jobs/maistra/xns-informer/maistra-xns-informer-maistra-2.4-presubmits.yaml
 
 Last time I saw that error was caused by a cluster migrate from build0x to build01

**Review by yxun — APPROVED** [review]

**Review by dgn — APPROVED** [review]
lgtm except for the branch regex

File: ci-operator/jobs/maistra/xns-informer/maistra-xns-informer-maistra-2.4-presubmits.yaml [hunk 1]
```diff
@@ -0,0 +1,100 @@
+presubmits:
+  maistra/xns-informer:
+  - agent: kubernetes
+    always_run: true
+    branches:
+    - ^maistra-2\.4$
+    - ^maistra-2\.4-
```
[the remark this answers is not in the document]
Comment (yxun) [reply ci-operator/jobs/maistra/xns-informer/maistra-xns-informer-maistra-2.4-presubmits.yaml:7]: well, that job yaml is generated by the Openshift CI side rendering. Usually we only write the ci-config yaml and then run `make update` for rendering the jobs directory yaml files.

[the remark this answers is not in the document]
Comment (yxun) [reply ci-operator/jobs/maistra/xns-informer/maistra-xns-informer-maistra-2.4-presubmits.yaml:7]: For example,
https://github.com/openshift/release/pull/33496/files#diff-b9f7b409d2a64f8dd8bdc7efa83db5f68354a0894ca9f38dd279c5e3aef0f4a7R34
that line in this PR defines the branch maistra-2.4, and then the jobs presubmits.yaml is automatically generating the branch regex

[the remark this answers is not in the document]
Comment (dgn) [reply ci-operator/jobs/maistra/xns-informer/maistra-xns-informer-maistra-2.4-presubmits.yaml:7]: okay ignore my comment in this case

**Pull request merged**